File: .emacs.d/ai-template/template.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
%file% - ????

???
 
"""
__author__  = "%name% <%mail%>"
__date__    = "%cdate%"
__credits__ = "Copyright (C) %cyear%, rakuten"
__version__ = "Version $Id$"

import sys , os , codecs , re , getopt , time 
sys.stdout = codecs.getwriter('utf-8')(sys.stdout)
sys.stderr = codecs.getwriter('utf-8')(sys.stderr)

#-----------------------------
# Default valuse
#-----------------------------
encode = 'utf-8'

#-----------------------------
# Message
#-----------------------------
def usage (code):
    """
    ???
    """
    msg = """
Usage: %file%  [options]
??????????????
Options:
 -h,--help:
            This message.
 -c,--charset  <charset>:
            charset.
"""
    print >>sys.stdout,msg
    sys.exit(code)


#-----------------------------
# Option parse
try:
    optlist, args = getopt.getopt(sys.argv[1:],"ht:f:s:b:e:E:c:",longopts=["help","to=","from=","subject=","body=","ext=","ext-binary=","charset="])
    for opt, args in optlist:
        # print opt
        if opt in ('-h','--help'):
            # debug_out(DEBUG_MSG,'help')
            usage(0)
        if opt in ('-c','--charset'):
            encode = args

except SystemExit:
    raise
except:
    usage(1)

print >>sys.stdout, 'encode       : ' + encode

#-----------------------------
# Class
class %cap%:
    """
    %cap% is the ?????
    """

    def __init__(self):
        """
        ???
        """
        pass

File: .emacs.d/ai-template/template.pl
#!/usr/bin/env perl
#---------------------------------------------
# %file% - ????
# 
#   Copyright (C) %cyear% rakuten 
#     by %name% <%mail%> 
#     Date : %cdate%
#---------------------------------------------

use strict;
use warnings;
use Getopt::Long;
use Pod::Usage;

use locale;
use POSIX qw(setlocale strftime mktime LC_ALL);

my $l    = setlocale(LC_ALL,"C");
my $d = strftime "%d/%b/%Y:%H:%M:%S", localtime( time() );

# use Dumpvalue;
# my $dmp = new Dumpvalue();

my %options = (
	       y   		  => 0,
	       debug              => undef,
	       );

# Option proc.
GetOptions(
	   'help' 		    => sub {pod2usage( -verbose => 3 , -exitval => 0 );},
	   'y'    		    => \$options{y},
	   'debug:n'   		    => \$options{debug},
	   ) or pod2usage( -verbose => 0 , -exitval => 1 );


__END__
=head1 NAME

%file% - ???? 

=head1 SYNOPSIS

%file%  [options] [???]

=head1 OPTIONS

=over 18

=item -y

Auto 'y' answar.

=item -help

View help document.

=back

=head1 TODO

=over 18

=item 

=back

=cut
